scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/authentication/ManageUsers.java

```diff
@@ -168,32 +173,57 @@ public static void manageUsers(String[] args) throws ManageUsersException {
             pubKeyPath = cmd.getOptionValue(KEYFILE_OPTION_NAME);
         }
 
+        if (cmd.hasOption(SOURCE_LOGINFILE_OPTION_NAME)) {
+            if (action == Action.DELETE) {
+                exitWithErrorMessage("Cannot use action delete with source login file.", null, null);
+            }
+            if (!cmd.hasOption(SOURCE_GROUPFILE_OPTION_NAME) && action == Action.CREATE) {
+                exitWithErrorMessage("Source group file must be provided when creating users with source login file.", null, null);
+            }
+            sourceLoginFilePath = cmd.getOptionValue(SOURCE_LOGINFILE_OPTION_NAME);
+            userInfo = null;
+        }
+        if (cmd.hasOption(SOURCE_GROUPFILE_OPTION_NAME)) {
+            if (action == Action.DELETE) {
+                exitWithErrorMessage("Cannot use action delete with source group file.", null, null);
+            }
+            if (!cmd.hasOption(SOURCE_LOGINFILE_OPTION_NAME) && action == Action.CREATE) {
+                exitWithErrorMessage("Source login file must be provided when creating users with source group file.", null, null);
+            }
+            sourceGroupFilePath = cmd.getOptionValue(SOURCE_GROUPFILE_OPTION_NAME);
+            userInfo = null;
+        }
+
         if (pubKeyPath == null) {
             pubKeyPath = getPublicKeyFilePath();
         }
+
         try {
             pubKey = Credentials.getPublicKey(pubKeyPath);
         } catch (KeyException e) {
             exitWithErrorMessage("Could not retrieve public key from '" + pubKeyPath, null, e);
         }
-
-        switch (action) {
-            case CREATE:
-                nonInteractive = (userInfo.getLogin() != null && userInfo.getPassword() != null && userInfo.getGroups() != null);
-                break;
-            case UPDATE:
-                nonInteractive = (userInfo.getLogin() != null && userInfo.getPassword() != null) || (userInfo.getLogin() != null && userInfo.getGroups() != null);
-                break;
-            case DELETE:
-                nonInteractive = userInfo.getLogin() != null;
-                break;
+        if (sourceLoginFilePath != null || sourceGroupFilePath != null) {
```

Comment:
Please extract all this new code used to define the value for _nonInteractive_ variable in a new method. Mainly for improving separation of concerns and thus enhancing readability and maintenance.